ledger/partial/ledger_test.go

```diff
@@ -20,9 +20,19 @@ import (
 
 func TestFunctionalityWithCompleteTrie(t *testing.T) {
 
-	l, err := complete.NewLedger(&fixtures.NoopWAL{}, 100, &metrics.NoopCollector{}, zerolog.Logger{}, complete.DefaultPathFinderVersion)
+	w := &fixtures.NoopWAL{}
+
+	l, err := complete.NewLedger(w, 100, &metrics.NoopCollector{}, zerolog.Logger{}, complete.DefaultPathFinderVersion)
 	require.NoError(t, err)
 
+	compactor := fixtures.NewNoopCompactor(l)
+	<-compactor.Ready()
+
+	defer func() {
+		<-l.Done()
+		<-compactor.Done()
+	}()
+
```

Comment:
what would happen if someone just initial the newLedger without the compactor ?

Reply:
If ledger doesn't have update activities, we don't need a compactor running.  Otherwise, `Ledger.Set` would block without compactor running.